test/python/circuit/test_random_circuit.py

```diff
@@ -81,3 +85,164 @@ def test_random_mid_circuit_measure_conditional(self):
             bool(getattr(instruction.operation, "condition", None)) for instruction in circ
         ]
         self.assertEqual([False, False, False, True], conditions)
+
+
+class TestRandomCircuitWithGraph(QiskitTestCase):
+    """Testing random_circuit_with_graph from
+    qiskit.circuit.random.utils.py"""
+
+    def setUp(self):
+        super().setUp()
+        n_q = 10
+        pydi_graph = rx.PyDiGraph()
+        pydi_graph.add_nodes_from(range(n_q))
+        # Some arbitrary coupling map
+        cp_map = [
+            (0, 2, None),
+            (1, 3, None),
+            (2, 4, None),
+            (3, 4, None),
+            (5, 7, None),
+            (4, 7, None),
+            (7, 9, None),
+            (5, 8, None),
+            (6, 9, None),
+        ]
+        pydi_graph.add_edges_from(cp_map)
+        self.interaction_graph = (pydi_graph, None, None, None)
+
+    def test_simple_random(self):
+        """Test creating a simple random circuit."""
+        circ = random_circuit_with_graph(interaction_graph=self.interaction_graph, depth=1)
+        self.assertIsInstance(circ, QuantumCircuit)
+        self.assertEqual(circ.width(), 10)
+
+    def test_min_times_qubit_pair_usage(self):
+        """Test at-least depth number of times qubit-pair is used."""
```

Comment:
the depth parameter specifies how often each qubit-pair must at least be used in a two-qubit gate before the circuit is returned